src/_posts/2021-04-18-crlf-vs-lf-normalizing-line-endings-in-git.md

```diff
@@ -79,164 +80,223 @@ It's easy to see why `CRLF` is redundant by today's standards—using both a car
+[In a previous section](#configuring-line-endings-in-git-with-coreautocrlf), we saw that you can achieve this desired behavior with `core.autocrlf=true`. The `.gitattributes` equivalent of this is [the `eol` attribute](https://git-scm.com/docs/gitattributes#_eol), which allows you to control which line ending gets applied in Git's working tree:
 
-{% include codeHeader.html %}
-```
-git ls-files --eol
+1. [`eol=lf`](https://git-scm.com/docs/gitattributes#Documentation/gitattributes.txt-Settostringvaluelf): converts to `LF` on checkout.
+2. [`eol=crlf`](https://git-scm.com/docs/gitattributes#Documentation/gitattributes.txt-Settostringvaluecrlf): converts to `CRLF` on checkout.
+
+In the case of batch scripts, we'd use `eol=crlf`:
+
+{% include codeHeader.html file: ".gitattributes" %}
+```bash
+# All files are checked into Git's index with LF
+* text=auto
+
+# These files are checked out using CRLF locally
+*.bat eol=crlf
 ```
 
-This will show you line ending information for all files that git is tracking, in a format like this:
+**This change won't take effect immediately**, so if you run `git ls-files --eol` after updating your `.gitattributes` file, you might still see `LF` line endings in the working tree. To update existing line endings in your working tree so they respect the `eol` attribute, you'll need to run the following set of commands [per this StackOverflow answer](https://stackoverflow.com/a/29888735/5323344):
 
+{% include codeHeader.html %}
 ```
-i/lf    w/crlf  attr/text=auto eol=lf   file.txt
+git rm --cached -r .
+git reset --hard
 ```
 
-From left to right, these are:
+{% aside %}
+  Since these commands do not alter Git's index, you will not see any changes staged for a commit. Think of this as "silently" refreshing the line endings in your working tree.
+{% endaside %}
 
-- `i`: The line endings in the index (what gets pushed to your repo). Should be `LF`.
-- `w`: The line endings in the working tree (may be `CRLF`, but that's okay if the index is `LF`).
-- `attr`: The `.gitattributes` rule that applies to this file.
-- The file name itself.
+You'll notice that this command differs from `git add --renormalize .`, which we previously used to update line endings in Git's *index*. Now, we're updating line endings in the *working tree* to reflect our `eol` preferences. If you now you run `git ls-files --eol`, you should see `i/lf w/crlf` for any files matching the specified pattern.
+
+{% aside %}
+  Under the hood, the `eol` attribute also implies `text` with no value. [Prior to Git 2.10.0](https://github.com/git/git/blob/master/Documentation/RelNotes/2.10.0.txt#L248), there was a bug in Git where `* text=auto eol=crlf` would imply `* text eol=crlf`—that is, the file-type-detection algorithm would not work. This has since then been fixed so that `text=auto` can safely be used together with `eol`.
```

Comment:
`* text=auto eol=crlf` is clear for me now, but it might confuse new readers. Maybe mention somewhere that `* text=auto eol=crlf` is identical to:

```
* text=auto
* eol=crlf
```